src/api/goods.ts

```diff
@@ -5,15 +5,22 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => response.json())
+    .catch(error => {
+      throw new Error(`Something went wrong ${error}`);
+    });
 }
 
-export const get5First = () => {
-  return getAll()
-    .then(goods => goods); // sort and get the first 5
+export const get5First = async () => {
+  const goods = await getAll();
+
+  return goods
+    .sort((a, b) => a.name.localeCompare(b.name))
+    .slice(0, 5);
 };
 
-export const getRedGoods = () => {
-  return getAll()
-    .then(goods => goods); // get only red
+export const getRedGoods = async () => {
+  const goods = await getAll();
+
+  return goods.filter(good => good.color === 'red');
```

Comment:
You just need color)
```suggestion
  return goods.filter(({ color })=> color === 'red');
```